src/busesApp/tests/buses.test.js

```diff
@@ -0,0 +1,132 @@
+import bcrypt from 'bcrypt';
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../app.js';
+import logger from '../../configs/winston.js';
+import DataSource from '../../data-source.js';
+import User from '../../models/user.js';
+
+chai.use(chaiHttp);
+chai.should();
+
+describe('buses router tests', () => {
+	before((done) => {
+		app.on('started', async () => {
+			try {
+				await DataSource.query('TRUNCATE "User" CASCADE');
+				await DataSource.query('TRUNCATE "Bus" CASCADE');
+			} catch (error) {
+				logger.error(error.message);
+			} finally {
+				done();
+			}
+		});
+	});
+	let token;
+	let createdBusId;
+	let busInfo;
+	it('should login an admin user', async () => {
+		const password = await bcrypt.hash('Andela@1234', 5);
+		const newUser = User.create({
+			email: 'test123@gmail.com',
+			password,
+			firstName: 'test',
+			lastName: 'Andela',
+			role: 'admin',
+		});
+		const user = await newUser.save();
+		const info = { email: user.email, password: 'Andela@1234' };
+		const response = await chai
+			.request(app)
+			.post('/api/v1/accounts/login')
+			.send(info);
+		expect(response).to.have.status(200);
+		token = response.body.data.access_token;
+	});
+	it('should return empty when no buses ', async () => {
+		const response = await chai
+			.request(app)
+			.get('/api/v1/buses')
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(200);
+		expect(response.body.data).to.have.length(0);
+	});
+	it('should create a new bus', async () => {
+		busInfo = {
+			plateNumber: 'RAD555E',
+			busType: 'Coaster',
+			seats: '27',
+		};
+		const response = await chai
+			.request(app)
+			.post('/api/v1/buses')
+			.send(busInfo)
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(201);
+		createdBusId = response.body.data.id;
+	});
+	it('should not create the same bus twice', async () => {
+		const response = await chai
+			.request(app)
+			.post('/api/v1/buses')
+			.send(busInfo)
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(409);
+	});
+	it('should get a single bus', async () => {
+		const response = await chai
+			.request(app)
+			.get(`/api/v1/buses/${createdBusId}`)
+			.set('Authorization', `Bearer ${token}`);
+		expect(response).to.have.status(200);
+	});
+	it('should update the bus', async () => {
+		const editInfo = {
+			plateNumber: 'RAD558E',
+			busType: 'Coaster',
+			seats: '27',
+		};
```

Comment:
have one editInfo object in that file created, unless the information is not the same